rideshare.rb

```diff
@@ -0,0 +1,136 @@
+rides = [
+  {
+    driver_id: "DR0004",
+    date: "3rd Feb 2016",
+    cost: 5,
+    rider_id: "RD0022",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0001",
+    date: "3rd Feb 2016",
+    cost: 10,
+    rider_id: "RD0003",
+    rating: 3,
+  },
+  {
+    driver_id: "DR0002",
+    date: "3rd Feb 2016",
+    cost: 25,
+    rider_id: "RD0073",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0001",
+    date: "3rd Feb 2016",
+    cost: 30,
+    rider_id: "RD0015",
+    rating: 4,
+  },
+  {
+    driver_id: "DR0003",
+    date: "4th Feb 2016",
+    cost: 5,
+    rider_id: "RD0066",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0004",
+    date: "4th Feb 2016",
+    cost: 10,
+    rider_id: "RD0022",
+    rating: 4,
+  },
+  {
+    driver_id: "DR0002",
+    date: "4th Feb 2016",
+    cost: 15,
+    rider_id: "RD0013",
+    rating: 1,
+  },
+  {
+    driver_id: "DR0003",
+    date: "5th Feb 2016",
+    cost: 50,
+    rider_id: "RD0003",
+    rating: 2,
+  },
+  {
+    driver_id: "DR0002",
+    date: "5th Feb 2016",
+    cost: 35,
+    rider_id: "RD0066",
+    rating: 3,
+  },
+  {
+    driver_id: "DR0004",
+    date: "5th Feb 2016",
+    cost: 20,
+    rider_id: "RD0073",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0001",
+    date: "5th Feb 2016",
+    cost: 45,
+    rider_id: "RD0003",
+    rating: 2,
+  },
+]
+
+# find the number of rides each driver has given
+# using @ allows me to shorthand use driver_rides in the other methods
+def number_of_rides(rides_array, driver_id)
+  @driver_rides = rides_array.select do |ride|
+    ride[:driver_id] == driver_id
+  end
+  return @driver_rides.length
+end
+
+# find the total amount of money each driver has made
+def money_earned
+  driver_earned = @driver_rides.sum do |ride|
+    ride[:cost]
+  end
+  return driver_earned
+end
+
+# find the average rating for each driver
+def avg_rating
+  sum_rating = @driver_rides.sum do |ride|
+    ((ride[:rating].to_f) / @driver_rides.length).round(2)
+  end
+  return sum_rating
+end
+
+# (optional) for each driver, which day did they make the most money
+def most_earnings_day
+  most_earning = @driver_rides.max_by do |ride|
+    ride[:cost]
+  end
+  return most_earning[:date]
+end
+
+# create an rides_array of driver ids, helpful to print statements below
+drivers = rides.map do |ride|
+  ride[:driver_id]
+end
+drivers = drivers.uniq.sort!
```

Comment:
Nice! I love this! This line is crucial -- it allowed you to get all of the data in a certain way in one line.

By the way, you don't need the `!` on `sort` because you re-assigned. :) You can either have the two syntaxes:

```ruby
drivers = drivers.uniq.sort
```

or

```ruby
drivers.uniq!.sort!
```